common.mk

```
@@ -1390,6 +1390,24 @@ $(srcdir)/ext/etc/constdefs.h: $(srcdir)/ext/etc/depend
 	$(exec) $(MAKE) -f - $(mflags) \
 		Q=$(Q) ECHO=$(ECHO) top_srcdir=../.. srcdir=. VPATH=../.. RUBY="$(BASERUBY)"
 
+KPEG = $(srcdir)/.bundle/bin/kpeg
+$(KPEG): main
+$(KPEG):
+	$(XRUBY) -C "$(srcdir)" bin/gem install --no-document \
+		--install-dir .bundle --conservative --silent "kpeg"
+
+generate-rdoc: $(srcdir)/lib/rdoc/markdown.rb $(srcdir)/lib/rdoc/markdown/literals.rb
+
+$(srcdir)/lib/rdoc/markdown.rb: $(srcdir)/lib/rdoc/markdown.kpeg $(KPEG)
+	$(ECHO) generating $@
+	$(XRUBY) -C "$(srcdir)" $(KPEG) -fs -o $(srcdir)/lib/rdoc/markdown.rb \
+		$(srcdir)/lib/rdoc/markdown.kpeg
```

To repair out-of-source build
```suggestion
	$(XRUBY) -C "$(srcdir)" $(KPEG) -fs -o lib/rdoc/markdown.rb \
		lib/rdoc/markdown.kpeg
```